src/core/device.c

```diff
@@ -907,7 +907,7 @@ static int device_dispatch_io(sd_device_monitor *monitor, sd_device *dev, void *
                 return 0;
         }
 
-        if (streq(action, "change"))
+        if (!STR_IN_SET(action, "add", "remove"))
```

Comment:
I found a nit to pick :-).

I think "move" (i.e. network device rename) should be treated as "add" for systemd device units.  We already get that right.  And we do not want to start running the "change" code for "move".  IOW this should be `if (!STR_IN_SET(action, "add", "move", "remove")))`

By the same strict logic, we should also have been checking "move" events for DEVPATH_OLD, and removing any matching device unit.  (But `/* We do not need swap_process_device_remove().  Block devices cannot be renamed, only network interfaces. */`).

Reply:
I'l have added "move" now to the list we check, but not the DEVPATH_OLD thing, that can be done in a separate PR I figure, if anyone really cares...